src/BaseTokenWrapper.sol

```diff
@@ -57,15 +57,21 @@ abstract contract BaseTokenWrapper is Ownable, IBaseTokenWrapper {
     uint16 referralCode,
     PermitSignature calldata signature
   ) external returns (uint256) {
-    IERC20WithPermit(TOKEN_IN).permit(
-      msg.sender,
-      address(this),
-      amount,
-      signature.deadline,
-      signature.v,
-      signature.r,
-      signature.s
-    );
+    try
+      IERC20WithPermit(TOKEN_IN).permit(
+        msg.sender,
+        address(this),
+        amount,
+        signature.deadline,
+        signature.v,
+        signature.r,
+        signature.s
+      )
+    {
+      //explicitly left blank to protect users from permit grifting
```

Comment:
you can move this comment just before the try.catch, and have the rest of the try-catch in 1 line.
seems cleaner to me